### PR TITLE
feat: add sanitizeBody, respond and sanitizeResponse implementation

### PR DESCRIPTION
**Pull request by VolodymyrWork23**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -70,22 +70,19 @@ type InitPreContextOut<T> = T extends HasInitPreContext<any, any>
   ? ReturnType<T['initPreContext']>
   : never;
```
Comment (captaincaius) [inline src/index.ts:72]: now sanitizeParams is different than the others...  I'd have preferred you did this in a separate issue/branch, but if you're really gonna do this in this issue, do it right:
1. change the code down there back to referencing this type (for consistency)
2. put these back, but change them to `type SanitizeParamsContextIn<T extends HasSanitizeParams<any, any>> = Parameters<T['sanitizeParams']>[0]`
3. Do the same to all the other *ContextIn and *ContextOut in this section (i.e. change them from being conditionals to simple types but with type constraints).
- [ ] bring back and make consistent

File: src/index.ts [hunk 2]
```diff
@@ -70,22 +70,19 @@ type InitPreContextOut<T> = T extends HasInitPreContext<any, any>
...
-type SanitizeParamsContextOut<T> = T extends HasSanitizeParams<any, any>
-  ? ReturnType<T['sanitizeParams']>
-  : never;
-
 type AttachDataIn<T> = T extends HasAttachData<any, any>
   ? Parameters<T['attachData']>[0]
   : never;
 type AttachDataOut<T> = T extends HasAttachData<any, any>
   ? PromiseResolveOrSync<ReturnType<T['attachData']>>
   : never;
```
Comment (captaincaius) [inline src/index.ts:79]: Seems you missed some like `TRight extends HasInitPreContext<any, any>` below - and also in all the other lifecycles.
- [ ] consistency in [workaroundery]
￼

File: src/index.ts (hunk 1 above)
Comment (captaincaius) [reply src/index.ts:72]: - [ ] ^ likewise with the newly added stages

**Comment by captaincaius** [comment]
reviewed - approved besides the little tasks left.  nice job finding a workaround for the exploding unions. :+1:

File: src/index.ts (hunk 2 above)
Comment (VolodymyrWork23) [reply src/index.ts:107]: done

File: src/index.ts (hunk 1 above)
Comment (VolodymyrWork23) [reply src/index.ts]: done

File: src/index.ts (hunk 2 above)
Comment (VolodymyrWork23) [reply src/index.ts:107]: done

File: src/index.ts [hunk 3]
```diff
@@ -63,47 +63,72 @@ export function NoopFinalAuth() {
...
+// https://github.com/Microsoft/TypeScript/issues/29368#issuecomment-453529532
+type PipedSanitizeParams<TLeft, TRight> = [TLeft] extends [
+  HasSanitizeParams<any, any>
+]
+  ? [TRight] extends [HasSanitizeParams<any, any>]
     ? HasSanitizeParams<
-        // @todo replace SanitizeParamsContextIn<TLeft> with Parameters<TLeft['sanitizeParams']>[0] and SanitizeParamsContextOut<TRight> with ReturnType<TRight['sanitizeParams']>
         SanitizeParamsContextIn<TLeft>,
         SanitizeParamsContextOut<TRight>
       >
```
Comment (captaincaius) [inline src/index.ts:134]: - [ ] [missed one]

Comment (VolodymyrWork23) [reply src/index.ts:134]: done

**Pull request merged**